templates/advanced-code.yml

```diff
@@ -0,0 +1,24 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  labels:
+    provider: codeready-toolchain
+    version: ${REVISION}
+  name: advanced-code
+objects:
+- apiVersion: v1
+  kind: Namespace
+  metadata:
+    annotations:
+      openshift.io/description: ${USERNAME}-code
+      openshift.io/display-name: ${USERNAME}-code
+      openshift.io/requester: ${USERNAME}
+    labels:
+      provider: codeready-toolchain
+      version: ${REVISION}
```

Comment:
I'm not sure if we should add the version here, because in the NsTemplateSet controller, there is also added the `revision` label. based on that label we will know if the namespace should be provisioned/updated or not. I know that it's different label - `version` vs `revision` but it can be very confusing... WDYT?

Reply:
ok, so we can remove it from the templates here, and add them at the `NSTemplateTier` level:
```
kind: NSTemplateTier
      metadata:
          name: basic
      spec:
    namespaces:
    - type: code
    	template: <yaml-with-ns-template>
    	revision: y8f907f6
    - type: dev
    	template: <yaml-with-ns-template>
    	revision: f8q907f4
    - type: stage
    	template: <yaml-with-ns-template>
    	revision: 907fy8f6
    - type: default
    	template: <yaml-with-ns-template>
    	revision: 23dfrt8
```
Later, the `NSTemplateSetController` will use this `revision` field from the `NSTemplateTier` to compare with the label set on the corresponding Namespace to determine if an update is needed. Is it ok for you?

Reply:
yup :+1:

Reply:
ok, so let me remove those fields here, then ;)

Reply:
done in e8ea902